closed/custom/modules/java.base/Copy.gmk

```diff
@@ -30,6 +30,28 @@ $(call openj9_copy_files,, \
 	$(OPENJ9_VM_BUILD_DIR)/include/jvmti.h \
 	$(INCLUDE_TARGET_DIR)/jvmti.h)
 
+ifeq ($(OPENJDK_TARGET_OS), zos)
+$(call openj9_copy_files,, \
+$(OPENJ9_TOPDIR)/runtime/include/jni_convert.h \
+$(INCLUDE_TARGET_DIR)/jni_convert.h)
+
+# Customized z/OS 31/64-bit Interoperability JNI header files.
+# These files are generated as part of OpenJ9 build.
+INCLUDE31_SOURCE_DIR := $(OPENJ9_VM_BUILD_DIR)/include31
+INCLUDE31_TARGET_DIR := $(INCLUDE_TARGET_DIR)/../include31
```

Comment:
Have you verified that using the name `include31` works with `jlink`?

Reply:
Thanks @keithc-ca.  This is a good point -- I hadn't realized jlink images include the headers.  I'll confirm.

Reply:
It's actually the `.jmod` files we are concerned about; they are inputs to `jlink` for producing (possibly custom) images. Some paths are treated specially in `.jmod` files (e.g. `classes/*`); we just need to play by the rules.

Reply:
Right. I see the following in the `java.base.jmod`, not immediately sure how to get additional directories to be included.

```sh
> jar tf java.base.jmod | grep include                                                                                                                                                                     
include/jni.h
include/jvmticmlr.h
include/zos/jni_md.h
include/ibmjvmti.h
include/jvmti.h
include/jni_convert.h
include/classfile_constants.h
```

Reply:
Looks like it's related to this: https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/make/CreateJmods.gmk#L52-L53  though not sure how `include` itself is specified for java.base.

Reply:
Looks like the module specific header files are staged into: ` $(SUPPORT_OUTPUTDIR)/modules_include/$(MODULE)` based on:

https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/openj9/make/common/modules/CopyCommon.gmk#L36-L63

Reply:
From what I can tell, jmod appears to support only `--header-files` and will stage files in the given directory into `include` directory of the output jmod file.  Looks like we'd have to modify jmod itself if we want a parallel `include31` directory.  Alternatively, we can have `include31` as a subdirectory within `include` folder by staging the files into modules_include target -- and just document the updated location in migration docs.   There will be another change  to compile /link commands of JNI natives for the libjvm31.x sidedeck file from Java 8, due to lib/s390x/j9vm -> lib/j9vm.   @keithc-ca @pshipton , thoughts?

Reply:
I think it would be simpler to put the new header files in a new subdirectory of `include`, e.g. `include/jni31` or similar. Applications that need to use the 31-bit shim API would compile with `{sdk}/include/jni31` on the include path (instead of `{sdk}/include31`).

Reply:
BTW, a `.jmod` file is not really in in `jar` format; you should use `jmod`:
```
$ ./bin/jmod list jmods/java.base.jmod | grep include | sort
include/classfile_constants.h
include/ibmjvmti.h
include/jni.h
include/jvmticmlr.h
include/jvmti.h
include/linux/jni_md.h
```

Reply:
Thanks @keithc-ca.  I got some initial feedback from 31/64-bit users... no one is against changing it to subdirectory of `include/` for Java 11+.  In some ways, `include/jni31` as you suggested is cleaner.   This simplifies this PR, as none of the `Image-pre.gmk` changes are needed.  I'll touch base with some other users who will be back from Thanksgiving next week to see if anyone raises additional concerns.

Reply:
Got feedback from users that the suggested change to include/jni31 was acceptable.  I'll update this PR with the necessary changes, and also address the other review comments.  Thanks.